services/api/src/resources/environment/harborSetup.js

```diff
@@ -0,0 +1,136 @@
+      var harborProjectID = res.body[0].project_id
+      logger.debug(`Got the harbor project id for project ${openshiftProjectName} successfully!`)
+    } catch (err) {
+      // 409 means project already exists
+      // 201 means project created successfully
+      if (err.statusCode == 404) {
+        logger.error(`Unable to get the harbor project id of "${openshiftProjectName}", as it does not exist in harbor!`)
+      } else {
+        logger.error(`Unable to get the harbor project id of "${openshiftProjectName}" !!`)
+      }
+    }
+
+    console.log(harborProjectID)
+    // Create robot account for new harbor project
+    try {
+      const res = await harborClient.post(`projects/${harborProjectID}/robots`, {
+        body: {
+          name: openshiftProjectName,
+          access: [
+            {
+              resource: `/project/${harborProjectID}/repository`,
+              action: "push"
+            }
+          ]
+        }
+      })
+      console.log(res)
+      var harborTokenInfo = res.body
+      console.log(harborTokenInfo)
+      logger.debug(`Robot was created for Harbor project ${openshiftProjectName} !`)
+    } catch (err) {
+      // 409 means project already exists
+      // 201 means project created successfully
+      if (err.statusCode == 409) {
+        logger.error(`Unable to create a robot account for harbor project "${openshiftProjectName}", as a robot account of the same name already exists!`)
+      } else {
+        logger.error(`Unable to create a robot account for harbor project "${openshiftProjectName}" !!`)
+      }
+    }
+
+    // Set Harbor env vars for lagoon environment
+    try {
+      await query(
+        sqlClient,
+        Sql.insertEnvVariable({
+          "name": "INTERNAL_REGISTRY_URL",
+          "value": lagoonHarborRoute,
+          "scope": "INTERNAL_CONTAINER_REGISTRY",
+          "environment": environmentID,
+        }),
+      );
+      logger.debug(`Environment variable INTERNAL_REGISTRY_URL for ${openshiftProjectName} created!`)
+
+    } catch (err) {
+      logger.error("Initial var already set!!")
+    }
+
+    try {
+      await query(
+        sqlClient,
+        Sql.insertEnvVariable({
+          "name": "INTERNAL_REGISTRY_USERNAME",
+          "value": harborTokenInfo.name,
+          "scope": "INTERNAL_CONTAINER_REGISTRY",
+          "environment": environmentID,
+        }),
+      );
+      logger.debug(`Environment variable INTERNAL_REGISTRY_USERNAME for ${openshiftProjectName} created!`)
+
+      await query(
+        sqlClient,
+        Sql.insertEnvVariable({
+          "name": "INTERNAL_REGISTRY_PASSWORD",
+          "value": harborTokenInfo.token,
+          "scope": "INTERNAL_CONTAINER_REGISTRY",
+          "environment": environmentID,
+        }),
+      );
+      logger.debug(`Environment variable INTERNAL_REGISTRY_PASSWORD for ${openshiftProjectName} created!`)
+    } catch (err) {
+      logger.error("Something went wrong setting env vars!", err)
```

Comment:
if you want to get the error, you need to output it like:
```
logger.error("Error while setting up harbor environment variables for ${openshiftProjectName}, error: ${err}")
```

Reply:
Updated.